src/TDebug.cpp

```diff
@@ -122,9 +156,117 @@ TDebug& TDebug::operator<<(const QList<QString>& list)
     }
     msg += "), ";
     return *this;
 }
+
+void TDebug::changeHostName(const Host* pHost, const QString& newName)
+{
+    if (pHost) {
+        QPair<QString, QChar>& pair = TDebug::smIdentifierMap[pHost];
+        pair.first = newName;
+    }
+}
+
+/* static */ void TDebug::addHost(Host* pHost)
+{
+    if (!pHost) {
+        return;
+    }
+    static const QString validIdentifiers{"123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"};
+    static const int maxIdCount = validIdentifiers.size();
+    int tryCount = -1;
+    do {
+        if (!TDebug::smUsedIdentifiers.contains(validIdentifiers.at(++tryCount))) {
+            break;
+        }
+    } while (tryCount < maxIdCount);
+
+    QString hostName = pHost->getName();
+    if (TDebug::smUsedIdentifiers.contains(validIdentifiers.at(tryCount))) {
+        // Run out of identifers - fall back to '?'
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, QLatin1Char('?'));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    } else {
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, validIdentifiers.at(tryCount));
+        TDebug::smUsedIdentifiers.insert(validIdentifiers.at(tryCount));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    }
+    TDebug localMessage(Qt::blue, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' started.\n").arg(hostName) >> nullptr;
+    TDebug tableMessage(Qt::white, Qt::black);
+    tableMessage << TDebug::displayNewTable() >> nullptr;
+}
+
+/* static */ void TDebug::removeHost(Host* pHost)
+{
+    TDebug::smIdentifierMap.remove(pHost);
+    TDebug localMessage(Qt::darkGray, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' ended.\n").arg(pHost->getName()) >> nullptr;
+    TDebug tableMessage(Qt::white, Qt::black);
+    tableMessage << TDebug::displayNewTable() >> nullptr;
+}
+
+/* static */ QString TDebug::displayNewTable()
+{
+    // We do not translate this currently as the Central Debug Console does not
+    // get translated content - yet?
+    QStringList messageLines;
+    QMapIterator<const Host*, QPair<QString, QChar>> itIdentifier(TDebug::smIdentifierMap);
+    while (itIdentifier.hasNext()) {
+        itIdentifier.next();
+        if (itIdentifier.key()) {
+            messageLines.append(QStringLiteral(" [%1] = \"%2\"").arg(itIdentifier.value().second, itIdentifier.value().first));
+        }
+    }
+    if (messageLines.count() > 1) {
+        std::sort(messageLines.begin(), messageLines.end());
+    }
+    messageLines.prepend(QStringLiteral(" [*] = System message, not belonging to a specific profile"));
+    if (TDebug::smIdentifierMap.count() > 1) {
+        messageLines.prepend(QStringLiteral("There are %1 profiles active now, so each message from a profile will be "
+                                            "prefixed with a single character inside a pair of '['...']'s as follows:").arg(TDebug::smIdentifierMap.count()));
```

Comment:
```suggestion
        messageLines.prepend(QStringLiteral("Now %1 profiles are active. Each message will be prefixed as follows:")
        .arg(TDebug::smIdentifierMap.count()));
```

Shorten message, drop obvious information.

Reply:
Guess *some* improvements could be made here.